### DISCUSSION TITLE
Allow Middleware like behavior  for api routes

### DISCUSSION THREAD

**Discussion by zhihengGet** [vercel/next.js#25779] [Ideas]
### Describe the feature you'd like to request

For all protected routes, say i put it under   /api/protected/  
 
I think it will be beneficial to make all the apis under it to go through a Authentication/authorization check by automatically pass them through a middleware, if middleware failed , then that request will be rejected.

Alternative would be using middleware handler similar to the middleware example but i think it is not very friendly 

Or next-connect to get express like routes but I think file route are more friendly....

### Describe the solution you'd like

> /api/protected/
> ---------------/middleware/authCheck.js
>  =================/apis 
> -----------------------------/apis that require authCheck.js

                                        
allow a file to be created under middleware folder, we will call it authCheck and all all folder under this middleware folder requires auth Check
 

### Describe alternatives you've considered

name the file that require check to be [....]hello.js

**Comment by mattcarlotta** [comment]
There is (or was) a [RFC for server/less middleware](https://github.com/vercel/next.js/issues/7208) that was [abandoned last year](https://github.com/vercel/next.js/pull/12218). Instead, if you want middleware, but don't want to use 3rd party packages, then you'll have to do something like [this](https://github.com/vercel/next.js/discussions/14415#discussioncomment-42291).
